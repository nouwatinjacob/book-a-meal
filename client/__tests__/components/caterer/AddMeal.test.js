import React from 'react';
import {
  shallow
} from 'enzyme';
import sweetalert from 'sweetalert';
import {
  AddMeal,
  mapStateToProps
} from '../../../src/components/caterer/AddMeal.jsx';

const props = {
  addMeal: () => Promise.resolve(),
  mealState: {},
  isLoading: true
};

jest.mock('sweetalert');

describe('AddMeal component', () => {
  localStorage.setItem('token', catererToken);
  const setup = () => shallow(<AddMeal {...props} />);

  it('should render AddMeal component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'onInputChange'
  when input value is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: 'Fried Rice',
        files: ['Fried-rice.jpg']
      }
    };

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onInputChange'
    );
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onInputChange'
  when input value is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: 'Fried Rice',
        files: ['Fried-rice.jpg']
      }
    };

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onInputChange'
    );
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onFormSubmit'
  when modify order button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    wrapper.setState({
      mealData: {
        name: 'Fried Rice',
        price: 2600,
        image: 'fried-rice.jpg'
      }
    });

    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'onFormSubmit'
    );
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalled();

    done();
  });
});
