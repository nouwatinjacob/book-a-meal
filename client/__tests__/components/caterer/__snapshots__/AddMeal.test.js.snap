// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMeal component should render AddMeal component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnHover={true}
      pauseOnVisibilityChange={true}
      position="top-right"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="login"
    >
      <div
        className="login-form"
      >
        <h3>
          Add New Meal
        </h3>
        <Loader
          color="#ff9600"
          height="50"
          margin="2px"
          type="Rings"
          visible={true}
          width="100"
        />
        <form
          encType="multipart/form-data"
          name="add-meal"
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Meal Name"
            type="text"
            value=""
          />
          <br />
          <input
            name="price"
            onChange={[Function]}
            placeholder="Meal Price"
            type="text"
            value=""
          />
          <br />
          <input
            accept="image/*"
            name="image"
            onChange={[Function]}
            placeholder="Meal Image"
            type="file"
          />
          <br />
          <br />
          <button
            className="button warning"
          >
            Add Meal
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
